src/main/java/ee/buerokratt/ruuter/helper/ScriptingHelper.java

```diff
@@ -25,20 +26,31 @@ public boolean containsScript(String s) {
         return Pattern.compile(SCRIPT_REGEX, Pattern.MULTILINE).matcher(s).find();
     }
 
-    public Map<String, Object> evaluateScripts(Map<String, Object> map, Map<String, Object> context, Map<String, Object> requestBody, Map<String, Object> requestParams) {
+    public Map<String, Map<String, Object>> evaluateRequestParameters(ConfigurationInstance ci, Map<String, Object> body, Map<String, Object> query, Map<String, Object> headers) {
```

Comment:
I don't know how "keen" i am on this new style, of moving all the param evaluation into this method.

Firstly i feel like it moves logic, that really should be handled by steps, into a helper - i don't feel like the scripting helper should need to consern itself with the exact specifics, of what input is required for most "http based" steps. Setting up either headers, body or query is really what the step itself should concern itself with, not the helper. Not to mention, that not all of these parameters are required for all steps - for example: you dont acutally need the body, when setting up data for get steps.

Secondly, this has resulted in a really weird return type. Generally, these kinds of return types, of "object maps (maps within maps)" are really to be avoided, and one should create a new class, to hold the necessary data. Now when it comes to that previous principle, i would say we have already "blurred" the lines a bit too much, and going forward we should cut back on using these kinds of objects (they the code a lot harder to read and understand).


The third thing, that annoys me with this, is that currently it made the code a bit shorter yes, but more difficult to read. Take the following example:

```
Map<String, Map<String, Object>> evaluatedParameters = ci.getScriptingHelper().evaluateRequestParameters(ci, args.getBody(), args.getQuery(), args.getHeaders());
return ci.getHttpHelper().doPost(args.getUrl(), evaluatedParameters.get("body"), evaluatedParameters.get("query"), ci.getMappingHelper().convertMapObjectValuesToString(evaluatedParameters.get("headers")));
```

Now first off, these lines are really long..., and getting the params from a map (and then even running operations on some) isn't really easy to understand either (people dont know what these maps may hold).

Lets make it a bit more readable:

```
Map<String, Map<String, Object>> evaluatedParameters = ci.getScriptingHelper().evaluateRequestParameters(ci, args.getBody(), args.getQuery(), args.getHeaders());
Map<String, Object> body = evaluatedParameters.get("body");
Map<String, Object> query = evaluatedParameters.get("query");
Map<String, String> headers = ci.getMappingHelper().convertMapObjectValuesToString(evaluatedParameters.get("headers"));
        
return ci.getHttpHelper().doPost(args.getUrl(), body, query, headers);
```
Now that it's more readable, one might wonder, would it be so different, to skip the `evaluateRequestParameters` method alltogether, and simply do the evaluation right here:

```
Map<String, Object> evaluatedBody = ci.getScriptingHelper().evaluateScripts(body, ci.getContext(), ci.getRequestBody(), ci.getRequestQuery(), ci.getRequestHeaders());
Map<String, Object> evaluatedQuery = ci.getScriptingHelper().evaluateScripts(query, ci.getContext(), ci.getRequestBody(), ci.getRequestQuery(), ci.getRequestHeaders());
Map<String, Object> evaluatedHeaders = ci.getScriptingHelper().evaluateScripts(headers, ci.getContext(), ci.getRequestBody(), ci.getRequestQuery(), ci.getRequestHeaders());
Map<String, String> mappedHeaders = ci.getMappingHelper().convertMapObjectValuesToString(evaluatedHeaders);

return ci.getHttpHelper().doPost(args.getUrl(), evaluatedBody, evaluatedQuery, mappedHeaders);
```
In the end we see that if we want to make the code readable, it isn't really that different to simply do the evaluation right at step level.

------------------------------------------

To sort of wrap up this "essay", my main concern isn't really with the idea of extracting "http parameter evaluation" into a new method (since it is done on many occasions), but the current solution doesnt really make things a lot better. I think there are two solutions here:
* move the "evaluation" straight into the steps - this will result in a bit of "copy/pasted" code, but atleast it will be easy to understand and keep step related logic in steps
* create a new object to return instead of the current `Map<String, Map<String, Object>>` and move the  `evaluateRequestParameters` method out of the helper . An actual object with properly named properties will give a clear overview of what can and can't be done with the return value (maybe the `evaluateRequestParameters` method could even be moved into said object, instead of the helper)